### PR TITLE
Support the bytes type in multilocale interop

### PR DESCRIPTION
**Pull request by dlongnecke-cray**
This PR adds support for the `bytes` type to multilocale libraries. It also refactors the code in `mli.cpp` to replace uses of `and` and `or` with `&&` and `||`.

---

Testing:

- [x] `ALL` on `linux64` when `CHPL_COMM=none`
- [x] `ALL` on `linux64` when `CHPL_COMM=gasnet` (no `cygwin`/`numpy`)
- [x] `test/interop/python/multilocale` on `linux64` when `CHPL_COMM=gasnet`
- [x] `test/interop/python` on `linux64` when `CHPL_COMM=gasnet`

### CODE REVIEW

**Comment by dlongnecke-cray** [comment]
I believe this is ready for review now.

**Comment by dlongnecke-cray** [comment]
Dealing with one test failures related to exported routines that are called internally without use of the `--library` flag at compilation time.

File: compiler/passes/resolveIntents.cpp [hunk 1]
```diff
@@ -275,7 +275,9 @@ void resolveArgIntent(ArgSymbol* arg) {
       // records/unions.
       bool addedTmp = (isRecord(arg->type) || isUnion(arg->type));
       FnSymbol* fn = toFnSymbol(arg->defPoint->parentSymbol);
-      if (fn->hasFlag(FLAG_EXTERN))
+      // TODO: Consider adding flag for export wrappers instead of this?
+      bool isExportAndLibrary = fLibraryCompile && fn->hasFlag(FLAG_EXPORT);
```
Comment (mppf) [inline compiler/passes/resolveIntents.cpp]: It seems to me this shouldn't depend on fLibraryCompile.

**Comment by bradcray** [comment]
Now that this is merged, I'm wondering if one of @lydia-duncan or @dlongnecke-cray (or someone else) would be able to pick up Lydia's Arkouda branch that uses `export`s to see whether we could handle the send/recv bytes cases via the bytes type (or perhaps we should switch all communications to use the bytes type rather than strings?).  Tagging @e-kayrakli as well since he's been working on bytes and UTF-8 validation, which is vaguely related.

I also think we should create (and ideally, do) some follow-up issues in which (a) strings can be communicated to a Python program without relying on strlen() (which I'm guessing would be similar to what we're doing for bytes here) and (b) strings show up as strings in Python code rather than bytes.

File: compiler/passes/resolveIntents.cpp (hunk 1 above)
Comment (dlongnecke-cray) [reply compiler/passes/resolveIntents.cpp]: It's true, it really shouldn't. The condition I'm really trying to test for here is "is this thing is an export wrapper", (as export wrappers are only created during `--library`).

Comment (dlongnecke-cray) [reply compiler/passes/resolveIntents.cpp]: I can write a patch that adds a "FLAG_EXPORT_WRAPPER", or something of the sort and uses that instead.

**Pull request merged**